feat(bsd): add CI testing using Kitchen's `vagrant` driver

myii opened the pull request:
### What does this PR do?

Add CI testing for FreeBSD (12.2 & 11.4) and OpenBSD (6.8) using Kitchen's `vagrant` driver.  These have been configured for local testing as well.

Already using these in the SaltStack Formulas org so proposing it here as well.

I'll use inline comments to mention any noteworthy points.

myii committed:
feat(bsd): add CI testing using Kitchen's `vagrant` driver

myii commented on kitchen.vagrant.yml:
```diff
@@ -0,0 +1,44 @@
+        sh -c 't=$(gshuf -i 1-15 -n 1); echo Sleeping $t seconds; sleep $t'
+
+verifier:
+  command: pytest --cache-clear -v -s -ra --log-cli-level=debug -k "not test_ping" tests/integration/
```
`test_ping` is actually working for local testing (on an Ubuntu 20.04 host).  It hangs until timeout in GitHub Actions, likely to be an issue with the macOS runner.  Haven't researched in great detail but these could be relevant:

* https://github.com/paramiko/paramiko/issues/721
* https://github.com/paramiko/paramiko/issues/735

myii commented on .github/workflows/kitchen.vagrant.yml:
```diff
@@ -0,0 +1,54 @@
+      fail-fast: false
+      matrix:
+        instance:
+          - py3-git-3000-freebsd-122
+          - py3-git-3000-freebsd-114
+          # - py3-git-3000-openbsd-68
+          - py3-git-3001-freebsd-122
+          - py3-git-3001-freebsd-114
+          # - py3-git-3001-openbsd-68
+          - py3-git-3002-freebsd-122
+          - py3-git-3002-freebsd-114
+          # - py3-git-3002-openbsd-68
+          - py3-git-master-freebsd-122
+          - py3-git-master-freebsd-114
+          # - py3-git-master-openbsd-68
+          - latest-freebsd-122
+          - latest-freebsd-114
+          - latest-openbsd-68
```
The commented OpenBSD instances aren't currently working, since the bootstrap script needs some updating.  I've included them so that it's easier to test any changes, if there are efforts made to fix that.

bryceml reviewed (changes requested):
Same as the windows tests, I'd prefer shellcheck and generate-actions-workflow run first and this not run if those don't pass.

myii commented:
Updated with the following:

* Rebased.
* Added `generate-actions-workflow` and `lint` (isn't automated, though).
* Added FreeBSD 13.0.
* Renamed `openbsd-68` to `openbsd-6`, since `generic/openbsd6` now provides OpenBSD 6.9 (which actually installs `3002.6`, as opposed to only `3001.1` on OpenBSD 6.8).
